pkg/operation/botanist/addons.go

```diff
@@ -235,13 +237,15 @@ func (b *Botanist) generateCoreAddonsChart(ctx context.Context) (*chartrenderer.
 		return nil, err
 	}
 
+	monitoringEnabled := helper.IsMonitoringEnabled(b.Config) && b.Shoot.Purpose != gardencorev1beta1.ShootPurposeTesting
```

Comment:
Now we have a lot of duplicated checks like this: "is monitoring enabled in gardenlet config" and "is shoot not of type purpose"?
This is likely to be regressed by future changes.

Can you deduplicate them by adding a function like `IsShootMonitoringEnabled` on `*Operation`?

Reply:
Is there some example on how to do this? I can't really follow

Reply:
ah I think I've found it. Hold on :D